ioc_copier/ioc_copier.py

```diff
@@ -140,13 +147,41 @@ def copy_folder(file_format, ioc_name):
     start_path = file_format.format(f"{ioc_name}-{padded_start_copy}")
     path = os.path.join(os.getcwd(), file_format.format(f"{ioc_name}-{padded_current_copy}"))
     try:
-        copytree(os.path.join(os.getcwd(), start_path), os.path.join(path))
+        copytree(os.path.join(os.getcwd(), start_path), os.path.join(path), ignore=ignore_patterns("st-*.cmd",
+                                                                                                   "build.mak",
+                                                                                                   "*.db",
+                                                                                                   "*.substitutions",
+                                                                                                   "*.req"))
     except FileExistsError:
         raise FileExistsError(f"Copy {padded_current_copy} already exists, please ensure that the initial copy "
                               f"argument is greater than the highest number IOC.") from None
     return path
 
 
+def generate_config(ioc):
+    """
+    Generate the config if copying ioc 01 as it should just reference ioc 01s config rather than duplicating it.
+    :param ioc: the ioc name
+    :return: the text lines of the config.
+    """
+    return ["<?xml version=\"1.0\" ?>\n",
+            "<ioc_config xmlns=\"http://epics.isis.rl.ac.uk/schema/ioc_config/1.0\" ",
+            "xmlns:xi=\"http://www.w3.org/2001/XInclude\">\n",
+            f"<xi:include href=\"../ioc{ioc}-IOC-01/config.xml\"  />\n",
+            "\n",
+            "</ioc_config>"]
+
+
+def comment_makefile(text):
+    """
+    comment out DB += lines from a makefile
+    :param text: the line to check whether to comment
+    :return: the line commented out if it is a DB+= line
+    """
+    text = [re.sub(r"(DB \+= )(?=.*\.db)", r"#DB += ", line) for line in text]
```

Comment:
In general we shouldn't like commented out code... Can we either make this delete the relevant lines, or tell the developer to delete them?

Reply:
Ok, I was just following the directions on the [wiki page](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Duplicating-IOCs) , im happy to change it to just deleteing the line